hackathon_site/dashboard/frontend/src/pages/Cart/Cart.test.tsx

```diff
@@ -146,4 +152,44 @@ describe("Cart Page", () => {
             expect(quantityDropdown).not.toHaveTextContent("3");
         });
     });
+
+    jest.mock("api/api", () => ({
+        ...jest.requireActual("api/api"),
+        post: jest.fn(),
+    }));
+
+    const mockedPost = post as jest.MockedFunction<typeof post>;
+    const orderUri = "/api/hardware/orders/";
+
+    const mockState: RootState = {
+        ...store.getState(),
+        [cartReducerName]: initialState,
+    };
+
+    type DispatchExts = ThunkDispatch<RootState, void, AnyAction>;
+    const mockStore = configureStore<RootState, DispatchExts>([thunk]);
+
+    test("Checks for correct POST request", async () => {
+        const response = makeMockApiListResponse(mockCartItems);
+        when(mockedPost).calledWith(orderUri).mockResolvedValue(response);
+
+        const store = mockStore(mockState);
+        const { getByTestId, getByText } = render(
+            <Provider store={store}>
+                <Cart />
+            </Provider>
+        );
+
+        const submitOrderBtn = getByTestId("submit-order-button");
+
+        fireEvent.click(submitOrderBtn);
+
+        await waitFor(() => {
+            expect(getByTestId("cart-quantity-total")).toHaveTextContent("0");
+            expect(getByText(/no items in cart/i)).toBeInTheDocument();
+            expect(mockedPost).toHaveBeenCalledWith(orderUri, {
+                hardware: [],
```

Comment:
This is a good point, you shouldn't be able to submit an empty ordchance. Could you implement a check for that and add a test for it? For this test, dispatch the addtocart action before submitting an order.

Reply:
Should the test for empty orders be in the cartSlice? That way I can test if the error snackbar shows up.

Reply:
It should be in cart.test.tsx because we should disable the submit order button in the ui

Reply:
Currently, in our Cart.test.tsx we don't actually disable the submit order button if there is no cart items, is that something I should add to the cart page?

Reply:
Yes, could you do that, it's shouldn't be a lot of work